### ISSUE TITLE
cfa_fit_plot does fail on DWLS Estimator

### ISSUE THREAD

**Issue by Buedenbender** [rempsyc/lavaanExtra#3]
Hey, regarding our last correspondence.
I create an example dataset, which runs fine with DWLS (and other Estimators) with the traditional lavaan see below

```r
# 1) Load Packages & Settings ---------------------------------------------
# install.packages("lavaanExtra", repos = c(
#   rempsyc = "https://rempsyc.r-universe.dev",
#   CRAN = "https://cloud.r-project.org"))
# remotes::install_github("Buedenbender/datscience")

rm(list=ls())
pacman::p_load(lavaanExtra,tidyverse, lavaan,skimr,datscience)

# Settings
set.seed(21)
N <-300 # sim. participants
k <- 10 # sim. variables

# 2) Simulate Likert-type Response format data ----------------------------
df <- tibble(ID = c(1:N))

# Create data for factor 1
for (i in c(1:round(k*0.7,0))) {
  df[paste0("item",i)] <- sample(1:5,N,replace=T, prob = c(.17,.5,.18,.1,.05)) 
}

# Create data for factor 2
for (i in c((round(k*0.7,0)+1):k)) {
  df[paste0("item",i)] <- sample(1:5,N,replace=T, prob = c(.0,.15,.25,.35,.25)) 
}

skim(df)
corstars(df)

# 3) Problem  -------------------------------------------------------------
# Create the model string
twofactor_model <- list(factor1 = paste0("item",c(1:6)),
                        factor2 = paste0("item",c(7:10)))

twofactor_model_lavaan <- write_lavaan(latent = twofactor_model)

# Fitting models with different estimators
for(esti in c("MLR","ML","DWLS","ULS","WLSM","WLSMV")){
  tmp_fit <- cfa(twofactor_model_lavaan,
                 data = df,
                 estimator = esti
  )
  assign(paste0("fit_",esti), value = tmp_fit)
}
rm(tmp_fit)

nice_fit(fit_ML,fit_MLR,fit_DWLS,fit_WLSM,fit_WLSMV,fit_ULS, nice_table = T)
```

Which works fine:
![grafik](https://user-images.githubusercontent.com/38650857/187021274-0156ed6a-e50a-4b1f-9933-f6ff7afc4936.png)

## Problem
However, the same loop over the estimators fails for cfa_fit_plot. As in our previous conversation, this example showcases why it is preferable not to default to PDF output in this case. Maybe it be worth to pickup an optional argument for the function to make at chooseable 

```r
# Remove all lavaan models
rm(list=grep("^fit_",ls(),value=T))
# Create the cfa models with cfa_fit_plot()
for(esti in c("MLR","ML","DWLS","ULS","WLSM","WLSMV")){
  tmp_fit <- cfa_fit_plot(twofactor_model_lavaan,
                 data = df,
                 estimator = esti
  )
  assign(paste0("lavvanExtra_fit_",esti), value = tmp_fit)
}

```

The loop cancels when it reaches DWLS (i.e., in my environment are three fit objects created with lavaanExtra)
![grafik](https://user-images.githubusercontent.com/38650857/187021525-64c748b2-ad51-473a-8a02-74f1ba388f09.png)

It might be however due to the ULS estimator that it breaks, here is the error from the console

```
Error in lav_options_set(opt) : 
  lavaan ERROR: missing="ml" is not allowed for estimator “uls”
In addition: Warning messages:
1: In lav_model_vcov(lavmodel = lavmodel, lavsamplestats = lavsamplestats,  :
  lavaan WARNING:
    The variance-covariance matrix of the estimated parameters (vcov)
    does not appear to be positive definite! The smallest eigenvalue
    (= -6.194973e-06) is smaller than zero. This may be a symptom that
    the model is not identified.
2: In lav_object_post_check(object) :
  lavaan WARNING: some estimated ov variances are negative
3: In lav_object_post_check(object) :
  lavaan WARNING: some estimated ov variances are negative
4: In lav_options_set(opt) :
  lavaan WARNING: information will be set to “expected” for estimator = “DWLS”
5: In lavaan::lavaan(model = model, data = data, estimator = estimator,  :
  lavaan WARNING:
    Model estimation FAILED! Returning starting values.
6: In lavaan::lavaan(model = model, data = data, estimator = estimator,  :
  lavaan WARNING: estimation of the baseline model failed.
7: In .local(object, ...) :
  lavaan WARNING: fit measures not available if model did not converge
```

## To Dos:

- [x] Make the output format (PDF, RStudio Device, ..) chooseable per argument / flag
- [x] Fix the errors for other estimators than ML, MLR

**Comment by rempsyc** [comment]
It is because cfa_fit_plot uses ‘missing = “fiml” per default to handle missing values, and that means full information maximum likelihood, so it would make sense that it doesn’t work with other methods that aren’t ML. I will make this argument optimal and this should fix this bug.

You can test it yourself by trying to make a regular CFA and using fiml for missing.

**Comment by Buedenbender** [comment]
Ok I can confirm adding missing = "fiml" makes traditional lavaan fail aswell.

**Comment by rempsyc** [comment]
Ok fixed in version `0.0.4`. It also opens the plot in RStudio Viewer instead of PDF per default, however, the `save.as.pdf` argument, when set to `TRUE`, exhibits the original behaviour.

Ref: https://lavaanextra.remi-theriault.com/news/index.html

``` r
library(lavaan)
#> This is lavaan 0.6-12
#> lavaan is FREE software! Please report any bugs.
library(lavaanExtra)
#> Suggested citation: Thériault, R. (2022). lavaanExtra: Convenience functions for lavaan (R package version 0.0.3) [Computer software]. https://lavaanExtra.remi-theriault.com/
(latent <- list(visual = paste0("x", 1:3),
                textual = paste0("x", 4:6),
                speed = paste0("x", 7:9)))
#> $visual
#> [1] "x1" "x2" "x3"
#> 
#> $textual
#> [1] "x4" "x5" "x6"
#> 
#> $speed
#> [1] "x7" "x8" "x9"

HS.model <- write_lavaan(latent = latent)
cat(HS.model)
#> ##################################################
#> # [---------------Latent variables---------------]
#> 
#> visual =~ x1 + x2 + x3
#> textual =~ x4 + x5 + x6
#> speed =~ x7 + x8 + x9
cfa_fit_plot(HS.model, HolzingerSwineford1939, estimator = "DWLS")
#> lavaan 0.6-12 ended normally after 44 iterations
#> 
#>   Estimator                                       DWLS
#>   Optimization method                           NLMINB
#>   Number of model parameters                        21
#> 
#>   Number of observations                           301
#> 
#> Model Test User Model:
#>                                                       
#>   Test statistic                                43.902
#>   Degrees of freedom                                24
#>   P-value (Chi-square)                           0.008
#> 
#> Model Test Baseline Model:
#> 
#>   Test statistic                               875.815
#>   Degrees of freedom                                36
#>   P-value                                        0.000
#> 
#> User Model versus Baseline Model:
#> 
#>   Comparative Fit Index (CFI)                    0.976
#>   Tucker-Lewis Index (TLI)                       0.964
#> 
#> Root Mean Square Error of Approximation:
#> 
#>   RMSEA                                          0.053
#>   90 Percent confidence interval - lower         0.027
#>   90 Percent confidence interval - upper         0.077
#>   P-value RMSEA <= 0.05                          0.401
#> 
#> Standardized Root Mean Square Residual:
#> 
#>   SRMR                                           0.059
#> 
#> Parameter Estimates:
#> 
#>   Standard errors                             Standard
#>   Information                                 Expected
#>   Information saturated (h1) model        Unstructured
#> 
#> Latent Variables:
#>                    Estimate  Std.Err  z-value  P(>|z|)   Std.lv  Std.all
#>   visual =~                                                             
#>     x1                1.000                               0.970    0.831
#>     x2                0.500    0.077    6.486    0.000    0.485    0.412
#>     x3                0.650    0.092    7.073    0.000    0.630    0.557
#>   textual =~                                                            
#>     x4                1.000                               1.001    0.860
#>     x5                1.057    0.115    9.224    0.000    1.058    0.820
#>     x6                0.951    0.102    9.297    0.000    0.952    0.869
#>   speed =~                                                              
#>     x7                1.000                               0.490    0.450
#>     x8                1.274    0.194    6.564    0.000    0.625    0.617
#>     x9                1.744    0.278    6.282    0.000    0.855    0.847
#> 
#> Covariances:
#>                    Estimate  Std.Err  z-value  P(>|z|)   Std.lv  Std.all
#>   visual ~~                                                             
#>     textual           0.412    0.052    7.908    0.000    0.424    0.424
#>     speed             0.226    0.037    6.042    0.000    0.476    0.476
#>   textual ~~                                                            
#>     speed             0.144    0.026    5.497    0.000    0.294    0.294
#> 
#> Variances:
#>                    Estimate  Std.Err  z-value  P(>|z|)   Std.lv  Std.all
#>    .x1                0.423    0.202    2.089    0.037    0.423    0.310
#>    .x2                1.152    0.135    8.560    0.000    1.152    0.831
#>    .x3                0.882    0.108    8.162    0.000    0.882    0.690
#>    .x4                0.352    0.180    1.959    0.050    0.352    0.260
#>    .x5                0.546    0.192    2.843    0.004    0.546    0.328
#>    .x6                0.294    0.171    1.721    0.085    0.294    0.245
#>    .x7                0.947    0.105    9.015    0.000    0.947    0.797
#>    .x8                0.635    0.129    4.908    0.000    0.635    0.620
#>    .x9                0.287    0.158    1.823    0.068    0.287    0.282
#>     visual            0.940    0.163    5.760    0.000    1.000    1.000
#>     textual           1.003    0.140    7.162    0.000    1.000    1.000
#>     speed             0.240    0.055    4.355    0.000    1.000    1.000
#> 
#> R-Square:
#>                    Estimate
#>     x1                0.690
#>     x2                0.169
#>     x3                0.310
#>     x4                0.740
#>     x5                0.672
#>     x6                0.755
#>     x7                0.203
#>     x8                0.380
#>     x9                0.718
#> lavaan 0.6-12 ended normally after 44 iterations
#> 
#>   Estimator                                       DWLS
#>   Optimization method                           NLMINB
#>   Number of model parameters                        21
#> 
#>   Number of observations                           301
#> 
#> Model Test User Model:
#>                                                       
#>   Test statistic                                43.902
#>   Degrees of freedom                                24
#>   P-value (Chi-square)                           0.008
```

<sup>Created on 2022-08-27 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

**Comment by Buedenbender** [comment]
Thanks! Closed by [this commit](https://github.com/rempsyc/lavaanExtra/commit/1d1c741ccc35b43df904bb5b3b44ae484f9799fa)
